### PR TITLE
Fix `/setspawn`. Also PR Scope Creep

### PR DESCRIPTION
**Pull request by sleepingdragoninn**
🎉 Okay so i was SUPPOSED to just fix `/setspawn` - I got a report that setting your spawn with `/setspawn` and switching caused you to get the "bed not found" message. So the `setSpawn` parameter just needed to be saved and reused in the module.

Cool, great idea - this PR ended up scope creeping to involve a lot of random cleanup because I wasn't happy with how long some of the extra files we made last PR were (sorry) - feel free to cut it down if there are any problems.

In summary:

 - `setSpawn` is now optionally saved with locations, fixing `/setspawn`-based spawns (don't require beds).
 - Tooltips for spawn and last location now show the dimension.
 - Location now knows how to serialize itself (getFloat() defaults to `0.0F` so pitch being 0 on spawns is preserved)
 - Location is a record (I deleted the interface. sorry)
 - Less code.

Apologies, I got nerdsniped.

### CODE REVIEW

**Comment by Draconizations** [comment]
Hahaha. It happens, I’ll review and release a new fix later today when class is over!

**Comment by Draconizations** [comment]
So guess who completely forgot about this. Merging and releasing now
